src/main/java/com/github/fge/jsonpatch/JsonPatchDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+
+/**
+ * JsonPatchDeserializer allows you to register a custom JsonPatchFactory for jackson to use
+ * during deserialization.
+ */
+public class JsonPatchDeserializer
+    extends JsonDeserializer<JsonPatch>
+{
+    private JsonPatchFactory factory;
+    public JsonPatchDeserializer(JsonPatchFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    @Override
+    public JsonPatch deserialize(final JsonParser jp,
+        final DeserializationContext ctxt)
+        throws IOException, JsonProcessingException
+    {
+        final JsonNode node = jp.readValueAsTree();
```

Comment:
The other thing I was thinking, what if we ditched the registry based parser and deserialization.  and just let everyone use basic jackson annotations for the Patch.  I'm not 100% sure if it would work, but basically take the old way of doing it and add new JsonPatch classes with Jackson annotations each time.  So we would add an ExtendedJsonPatch class and just restate all of the operations in it.